test/compress/issue-1609.js

```diff
@@ -0,0 +1,56 @@
+chained_evaluation_1: {
+    options = {
+        collapse_vars: true,
+        evaluate: true,
+        reduce_vars: true,
+        unused: true,
+    }
+    input: {
+        (function() {
+            var a = 1;
+            (function() {
+                var b = a, c;
+                c = f(b);
+                c.bar = b;
+            })();
+        })();
+    }
+    expect: {
+        (function() {
+            (function() {
+                var c;
+                c = f(1);
+                c.bar = 1;
+            })();
+        })();
+    }
+}
+
+chained_evaluation_2: {
+    options = {
+        collapse_vars: true,
+        evaluate: true,
+        reduce_vars: true,
+        unused: true,
+    }
+    input: {
+        (function() {
+            var a = "long piece of string";
+            (function() {
+                var b = a, c;
+                c = f(b);
+                c.bar = b;
+            })();
+        })();
+    }
+    expect: {
+        (function() {
+            var a = "long piece of string";
+            (function() {
+                var c;
+                c = f(a);
```

Comment:
This looks odd. `collapse_vars` would not make this substitution of `f(b)` -> `f(a)` because function `f` is undeclared and would be a side effect. There might be still another clone issue in `reduce_vars` in `var b` of value `a`.

v2.7.5 by comparison produces:
```
$ uglifyjs chained_evaluation_2.js -c collapse_vars,reduce_vars -b
WARN: Replacing constant a [chained_evaluation_2.js:4,24]
!function() {
    !function() {
        var c, b = "long piece of string";
        c = f(b), c.bar = b;
    }();
}();
```